tools/wptrunner/wptrunner/executors/executorcontentshell.py

```diff
@@ -60,6 +61,16 @@ def do_test(self, command, timeout=None):
         deadline = time() + timeout if timeout else None
         # The first block can also contain audio data but not in WPT.
         text = self._read_block(deadline)
+
+        if type(text) == str and text.startswith('#LEAK'):
```

Comment:
Can text be another type?

Reply:
I don't think text can be another type in a valid response, but it "might" be None in case we get a error response, or a late response.
So as as far as I can see #LEAK appears to always be at the start based on the ci/try runs I've done.
Currently when it leaks it fails anyways because text cannot be parsed, so this should be an improvement.

If #LEAK does appear after I can make a fix for that, but I don't think that should be the case.

Reply:
Not sure if you notice this, anything thoughts on this?

Reply:
I tried that at first, but since we're using mozlog we won't be able to have a LEAK result type without some extra work I believe. 

I'll make the update to check is not none, 

As per RWT, I did notice for RWT it's consistently AFTER the text portion, but for WPT it's always coming before the text.
Coming before the text causes some issues because right now wptrunner assumes that text block will be the first thing to come, but it is not the case with #LEAK, not sure why. 

crbug.com/1448902 goes over this difference.

Another major thing is that in RWT we're always checking inside the stderr for crash/leak, while in wpt for crash we're checking inside stdout. We could do the same here for LEAK and just look inside stderr, but we would still need to handle the LEAK coming in front of stdout else we'll see an error when it tries to parse text as a result

Reply:
How about the change below?

change
```
def _read_line(io_queue, deadline=None, encoding=None, errors="strict", raise_crash=True):
```
 to
```
def _read_line(io_queue, deadline=None, encoding=None, errors="strict", raise_crash_leak=True):
```

Then add below to _read_line
```
        if raise_crash_leak and line.startswith(b"#LEAK"):
            raise LeakError()
```

Then in _convert_exception, add
```
    if isinstance(exception, CrashError):
        test.restart_test_after = True
        return (test.result_cls("ERROR", errors), [])
```

Reply:
Done, updated to raise leak error, and return error as result